src/decorators/registry.ts

```diff
@@ -0,0 +1,25 @@
+import { handleDecorator } from './handleDecorator';
+import { RegistryItem } from '../Registry';
+
+export interface RegistryConfig {
+	[ name: string ]: RegistryItem;
+}
+
+/**
+ * Decorator that can be used to register a widget with the registry
+ */
+export function registry(nameOrConfig: string | RegistryConfig, loader?: RegistryItem) {
```

Comment:
Do you think this would be better overloaded rather than with an optional type? For example if the first argument is `RegistryConfig` you shouldn't be able to pass a `loader` argument?